pkg/kubelet/kubelet.go

```diff
@@ -432,6 +433,10 @@ func (kl *Kubelet) Run(updates <-chan PodUpdate) {
 	if kl.dockerPuller == nil {
 		kl.dockerPuller = dockertools.NewDockerPuller(kl.dockerClient, kl.pullQPS, kl.pullBurst)
 	}
+	if kl.dockerCache == nil {
+		kl.dockerCache = dockertools.NewDockerCache(kl.dockerClient)
+	}
+	go util.Forever(func() { kl.dockerCache.UpdateContainers() }, 50*time.Millisecond)
```

Comment:
I was thinking that when the cache gets a request, if its cached results have not aged out it returns that, else it will make the request (synchronously) and return the results.

We can get more fancy by starting the async updates on the first request and if we don't get another request for TIMEPERIOD we stop the async updates. That may not be necessary (although doesn't seem that bad once I wrote it down :) )